controller/controller.go

```diff
@@ -206,6 +253,11 @@ func (c controller) GetStatus(req Request) ([]fleet.UnitStatus, error) {
 	return status, maskAny(err)
 }
 
+func (c controller) WaitForTask(taskObject *task.TaskObject, closer <-chan struct{}) (*task.TaskObject, error) {
```

Comment:
That would mean to change the interface within the controller and the task service. I did it as it is on purpose. So all the time you deal with the task object. IMO this is already nice.